components/clock.rst

```diff
@@ -193,6 +198,49 @@ being in a month or another.
 
     The :class:`Symfony\\Component\\Clock\\ClockAwareTrait` was introduced in Symfony 6.3.
 
+.. _clock_date-point:
+
+The ``DatePoint`` Class
+-----------------------
+
+When using the Clock component, you manipulate instances of
+:class:`Symfony\\Component\\Clock\\DatePoint`. The main purpose of this
+class is is that it fully leverages the Clock component. Indeed, when creating a new
+``DatePoint`` object, the date and time are being fetched
+from the :class:`Symfony\\Component\\Clock\\Clock` class. This means
+that if you did any changes to the clock as stated in the
+:ref:`usage section <clock_usage>`, it will be reflected when creating
+a new ``DatePoint``. This can be particularly useful when
+:ref:`writing time-sensitive tests <clock_writing-tests>`.
+
+Creating a new ``DatePoint`` is as straightforward as creating other date
+objects::
+
+    use Symfony\Component\Clock\DatePoint;
+
+    // create a DatePoint set at the current date and time
+    $now = new DatePoint();
+
+    // you can specify a timezone
+    $withTimezone = new DatePoint(timezone: new \DateTimezone('UTC'));
+
+    // you can also create a DatePoint from a reference DateTimeImmutable
+    $immutable = new \DateTimeImmutable();
+    $fromImmutable = new DatePoint(reference: $immutable);
+
+Also, ``DatePoint`` offers a stricter alternative to ``DateTimeImmutable`` by
+having stricter error handling, as well as stricter return types. ``DatePoint``
+actually extends ``DateTimeImmutable``, which means you can widely use it in
+your application if you already type against ``DateTimeImmutable`` across your
+code base.
```

Comment:
```suggestion
``DatePoint`` extends ``DateTimeImmutable`` so that you can use it seamlessly everywhere a ``DateTimeImmutable`` is needed. In addition ``DatePoint`` offers stricter return types and provides consistent error handling across versions of PHP, thanks to polyfilling PHP 8.3's behavior on the topic.

Reply:
we could link to https://wiki.php.net/rfc/datetime-exceptions on the last sentence maybe?

Reply:
That's a great idea, I added the link